yggdrasil_rebirth_engine.txt

```diff
@@ -0,0 +1,11 @@
+Name:Yggdrasil, Rebirth Engine
+ManaCost:3
+Types:Legendary Artifact
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChangeZoneAll | TriggerDescription$ When CARDNAME enters the battlefield, exile all creature cards from your graveyard.
+SVar:TrigChangeZoneAll:DB$ ChangeZoneAll | ChangeType$ Card.YouOwn | Origin$ Graveyard | Destination$ Exile | SubAbility$ DBChooseCard
+A:AB$ Dig | Cost$ T | Defined$ You | DigNum$ 3 | ChangeNum$ All | DestinationZone$ Exile | SpellDescription$ Exile the top three cards of your library.
+A:AB$ ChooseCard | Cost$ 4 T | Choices$ Creature.ExiledWithSource | ChoiceZone$ Exile | SorcerySpeed$ True | SubAbility$ DBChangeZone | SpellDescription$ Put a creature card exiled with Yggdrasil onto the battlefield under your control. It gains haste until end of turn. Activate only as a sorcery.

```

Comment:
don't see why extra ChooseCard effect is needed

Reply:
why resolve, still needs change?

Reply:
Hm, you mean that it could be just ChangeZone without ChooseCard? I might have understood it wrongly.

Reply:
yea I meant it like that